packages/engine-server/src/util/inMemoryNoteCache.ts

```diff
@@ -0,0 +1,19 @@
+import { NoteProps } from "@dendronhq/common-all";
+import { InMemoryNoteCacheImpl } from "./impl/inMemoryNoteCacheImpl";
+
+/**
+ * In memory cache for {@link NoteProps}.
+ *
+ * Use {@link InMemoryNoteCacheFactory} to get an instance of this interface.*/
+export interface InMemoryNoteCache {
+  /** Returns list of {@link NoteProps} that have matching file name (ignoring
+   *  the file name case). Will return empty list if no notes match.  */
+  getNotesByFileNameIgnoreCase(fileName: string): NoteProps[];
+}
+
+/** Factory for {@link InMemoryNoteCache} */
+export class InMemoryNoteCacheFactory {
```

Comment:
Perhaps a Factory isn't necessary here - are we actually going to be able to return a different implementation based on the argument (`NoteProps`)?  Or are we injecting the factory into a class such that the class can control object lifecycle?  Seems like no in either case.